frontend/src/templates/Field/Date/DateField.tsx

```diff
@@ -61,6 +63,7 @@ export const DateField = ({
         control={control}
         name={schema._id}
         rules={validationRules}
+        defaultValue={isMyInfoFormField(schema) ? schema.fieldValue : ''}
```

Comment:
i don't think we should inject `defaultValues` into the fields like this. This may result in having to update these fields whenever something else changes, especially if this change is solely for admin form builder (and fields are also used in public forms). 

Since all fields are wrapped by `react-hook-form`'s `FormProvider` by design, should we update the `defaultValue`s via its initial `useForm` call instead?

The form methods are passed down in [`FieldRowContainer.tsx`](https://github.com/opengovsg/FormSG/blob/form-v2/develop/frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FieldRow/FieldRowContainer.tsx#L99), and you may find out how public form fields augment their MyInfo values in [`FormFields.tsx`](https://github.com/opengovsg/FormSG/blob/form-v2/develop/frontend/src/features/public-form/components/FormFields/FormFields.tsx#L36)

Reply:
good call - this seems way superior overall. will change to initializing the values in `FieldRowContainer` rather than hacking around